ShopAppTests/App/Account/Providers/OrderListTableProviderSpec.swift

```diff
@@ -0,0 +1,105 @@
+            tableView = UITableView()
+            tableView.registerNibForCell(CheckoutCartTableViewCell.self)
+            tableView.dataSource = tableProvider
+            tableView.delegate = tableProvider
+        }
+        
+        describe("when provider created") {
+            it("should return correct count") {
+                let sectionsCount = tableProvider.numberOfSections(in: tableView)
+                expect(sectionsCount) == 0
+                
+                let rowsCount = tableProvider.tableView(tableView, numberOfRowsInSection: 0)
+                expect(rowsCount) == 1
+                
+                let header = tableProvider.tableView(tableView, viewForHeaderInSection: 0)
+                expect(header).to(beNil())
+                
+                let footer = tableProvider.tableView(tableView, viewForFooterInSection: 0)
+                expect(footer).to(beNil())
+            }
+        }
+        
+        describe("when orders did set") {
+            var order: Order!
+            
+            beforeEach {
+                order = Order()
+                order.id = "order id"
+                order.number = 1000
+                order.createdAt = Date()
+                order.currencyCode = "USD"
+                order.totalPrice = Decimal(floatLiteral: 10)
+                tableProvider.orders = [order, order, order]
+                tableView.reloadData()
+            }
+            
+            it("should return correct count") {
+                let sectionsCount = tableProvider.numberOfSections(in: tableView)
+                expect(sectionsCount) == tableProvider.orders.count
+                
+                let header = tableProvider.tableView(tableView, viewForHeaderInSection: 0)
+                expect(header).to(beAnInstanceOf(OrderHeaderView.self))
+                
+                let footer = tableProvider.tableView(tableView, viewForFooterInSection: 0)
+                expect(footer).to(beAnInstanceOf(OrderFooterView.self))
+                
+                let headerHeight = tableProvider.tableView(tableView, heightForHeaderInSection: 0)
+                expect(headerHeight) == kOrderHeaderViewHeight
+                
+                let footerHeight = tableProvider.tableView(tableView, heightForFooterInSection: 0)
+                expect(footerHeight) == kOrderFooterViewHeight
+            }
+            
+            it("should return correct cell class") {
+                let indexPath = IndexPath(row: 0, section: 0)
+                let cell = tableProvider.tableView(tableView, cellForRowAt: indexPath)
+                expect(cell).to(beAnInstanceOf(CheckoutCartTableViewCell.self))
+            }
+            
+            it("should select order") {
+                let providerDelegateMock = OrdersListTableProviderMock()
+                tableProvider.delegate = providerDelegateMock
+                expect(providerDelegateMock.selectedOrder).to(beNil())
```

Comment:
You can don't check any variables after mock's init and before some changes or actions in provider. It doesn't make sense.